libnetwork/controller_linux.go

```diff
@@ -0,0 +1,33 @@
+package libnetwork
+
+import (
+	"github.com/docker/docker/libnetwork/iptables"
+	"github.com/docker/docker/libnetwork/netlabel"
+	"github.com/docker/docker/libnetwork/options"
+)
+
+// enabledIptablesVersions returns the iptables versions that are enabled
+// for the controller.
+func (c *Controller) enabledIptablesVersions() []iptables.IPVersion {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.cfg == nil {
+		return nil
+	}
+	// parse map cfg["bridge"]["generic"]["EnableIPTable"]
```

Comment:
Yeah, ISTR that "disable iptables" only really worked for the default bridge, because it was still needed for overlay (and could not be disabled for that).

There's a lot of weird stuff in this area (overall as well, controller calling iptables rule, which then calls the controller to remove itself .... etc.

Reply:
Yeah, iptables can't be disabled for the overlay driver, or it wouldn't work otherwise.